server/zanata-war/src/main/java/org/zanata/adapter/JsonAdapter.java

```diff
@@ -46,4 +48,9 @@ protected void updateParamsWithDefaults(IParameters params) {
         Parameters p = (Parameters) params;
         p.fromString("extractIsolatedStrings.b=true");
     }
+
+    @Override
+    protected String getIdFor(TextUnit tu, String content, String subDocName) {
+        return HashUtil.generateHash(tu.getName().concat(content));
```

Comment:
Hmm.

Is that the local key, eg `title`, or some sort of global key, eg `test.test1.title` for a nested element? Global would be better.

I suppose any migration problems can be resolved by re-uploading the source and re-translating, although that could be a lot of work. Until then, a lot of translations would be missing when downloading.

As long as we still have the json file in the file system, we could implement a data migration which reparsed the json file and copied the translations to the new TextFlows. That's a lot of work though, considering we don't use json translations ourselves.